### PR TITLE
Refactor utility to calculate ensemble mean and recenter:

### PR DESCRIPTION
**Pull request by hu5970**
1)  Use all cores to read; each core reads one or several 2D fields parallely from one member
    2)  read member one by one; calculate mean and perturbation in each core; update member one by one
    3)  file needs to have right chunk size (2D dimension).

### CODE REVIEW

**Comment by TingLei-NOAA** [comment]
@hu5970  my first question is about the verification of the results from the new recenter tool.  The new tool has generated the identical results as the "old " one,  right?

File: ens_mean_recenter/module_mpi_arrange.f90 [hunk 1]
```diff
@@ -142,174 +142,139 @@ subroutine arrange(this,ncfs_dyn,ncfs_tracer,ncfs_sfc)
...
-    write(6,*) 'total 2d level=',nlvl2d,' 3d level=',nlvl3d
+    write(6,*) 'total 2d_level=',nlvl2d,' 3d_level=',nlvl3d, &
+               ' 3d_level_small =',nlvl3d_small
     write(6,*) 'max level =',nlvlmax
 !
 !  check if we have enought cores
 !
-    if( ntotalcore-nlvl2d < nlvl3d ) then
-        write(6,*) 'not enough cores for the paralleli IO',ntotalcore-nlvl2d,nlvl3d
+    if( ntotalcore-nlvl2d-nlvl3d_small < nlvl3d ) then
```
Comment (TingLei-NOAA) [inline ens_mean_recenter/module_mpi_arrange.f90:179]: Why do we need such a criteria?

Comment (hu5970) [reply ens_mean_recenter/module_mpi_arrange.f90:179]: There are two 3D arrays, 65 level atmosphere and 9 level soil levels. I treat 9 level soil as single level fields (nlvl3d_small).

**Comment by hu5970** [comment]
New and old code will not product bit by bit identical results. But the difference between them should within last digit of the range of a real 4 number represent.

File: ens_mean_recenter/module_mpi_arrange.f90 [hunk 2]
```diff
@@ -142,174 +142,139 @@ subroutine arrange(this,ncfs_dyn,ncfs_tracer,ncfs_sfc)
...
-              nx(nn)=ncfs_sfc%dim_1(i)
-              ny(nn)=ncfs_sfc%dim_2(i)
+              varname(nn)=trim(ncfs_all%list_varname(i))
+              vartype(nn)=ncfs_all%vartype(i)
+              nx(nn)=ncfs_all%dim_1(i)
+              ny(nn)=ncfs_all%dim_2(i)
               lvlbegin(nn)=1
               lvlend(nn)=1
-              fileid(nn)=3
+              fileid(nn)=myfieldid
```
Comment (TingLei-NOAA) [inline ens_mean_recenter/module_mpi_arrange.f90:268]: Finally, shall we expect the final nn=ntotalcore?  or if it is acceptable if it is a little smaller than ntotalcore (that means some cores are not used)?

Comment (hu5970) [reply ens_mean_recenter/module_mpi_arrange.f90:268]: nn can be smaller than ntotalcore.

**Review by TingLei-NOAA — APPROVED** [review]

**Review by guoqing-noaa — APPROVED** [review]

**Review by chunhuazhou — APPROVED** [review]

**Pull request merged**